pkg/orchestration/wiring/saml/plugin.go

```diff
@@ -0,0 +1,136 @@
+package saml
+
+import (
+	"encoding/json"
+	"fmt"
+
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager"
+	orch_v1 "github.com/atlassian/voyager/pkg/apis/orchestration/v1"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/knownshapes"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/osb"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	clusterServiceClassExternalID                      = "277d87d1-4cf6-4d96-8bd6-3affb551f21c"
+	clusterServicePlanExternalID                       = "55beb5f4-16f0-4fe5-abcb-95e077271bde"
+	ResourceType                  voyager.ResourceType = "SAML"
+	ResourcePrefix                                     = "SAML"
+)
+
+type autowiringOnlySpec struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	URL         string `json:"url"`
+}
+type finalSpec struct {
+	autowiringOnlySpec
+}
+type WiringPlugin struct {
+}
+
+func New() *WiringPlugin {
+	return &WiringPlugin{}
+}
+
+func (p *WiringPlugin) WireUp(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) wiringplugin.WiringResult {
```

Comment:
If we don't need anything in the WiringPlugin I believe we can just expose the `WireUp` function, see the SQS resource.

Reply:
I think we'd want to make plugins configurable i.e. to have constructor functions regardless of whether they need any parameters or not. So I think it's fine both ways as we'll likely need to refactor this anyway.